x-pack/plugins/file_upload/public/components/geo_upload_form/geo_upload_form.tsx

```diff
@@ -119,13 +141,39 @@ export class GeoUploadForm extends Component<Props, State> {
         <GeoFilePicker onSelect={this._onFileSelect} onClear={this._onFileClear} />
         {this._renderGeoFieldTypeSelect()}
         {this.state.hasFile ? (
-          <IndexNameForm
-            indexName={this.props.indexName}
-            indexNameError={this.props.indexNameError}
-            onIndexNameChange={this.props.onIndexNameChange}
-            onIndexNameValidationStart={this.props.onIndexNameValidationStart}
-            onIndexNameValidationEnd={this.props.onIndexNameValidationEnd}
-          />
+          <>
+            <IndexNameForm
+              indexName={this.props.indexName}
+              indexNameError={this.props.indexNameError}
+              onIndexNameChange={this.props.onIndexNameChange}
+              onIndexNameValidationStart={this.props.onIndexNameValidationStart}
+              onIndexNameValidationEnd={this.props.onIndexNameValidationEnd}
+            />
+            <EuiSpacer size="m" />
+            <EuiFormRow
+              label={
+                <EuiToolTip
+                  anchorClassName="eui-alignMiddle"
+                  content={i18n.translate('xpack.fileUpload.uploadSize.tooltip', {
+                    defaultMessage: 'Use small upload size to alleviate request timeout failures.',
```

Comment:
Choosing "Small" on a fast connection could cause the upload to take longer, right? Maybe we should make this known to the user somehow.

```suggestion
                    defaultMessage: 'Normal will upload files as quickly as possible. Use small upload size to alleviate request timeout failures.',
```